app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt

```diff
@@ -93,10 +93,10 @@ class EmbarkActivity : BaseActivity(R.layout.activity_embark) {
     private fun showNextView(embarkData: EmbarkModel, passage: EmbarkStoryQuery.Passage?) {
         val offerKeys = embarkData.passage?.offerRedirect?.data?.keys
         if (offerKeys != null && offerKeys.isNotEmpty()) {
-            val offerIds = offerKeys.map { key -> key?.let { model.getFromStore(it) } }
+            val offerIds = offerKeys.mapNotNull { key -> key?.let { model.getFromStore(it) } }
             showWebOffer(offerIds)
         } else if (embarkData.passage?.externalRedirect?.data?.location == EmbarkExternalRedirectLocation.OFFER) {
-            val key = model.getFromStore("quoteId")
+            val key = model.getFromStore("quoteId") ?: ""
```

Comment:
I think it might be warranted to error out here. Do you agree?

Reply:
Totally agree, fixing